src/components/numbered-code-snippet/numbered-code-snippet.js

```diff
@@ -0,0 +1,323 @@
+    const displayCode = props.highlightedCode || props.code;
+    const splitDisplayCode = displayCode.trim().split('\n');
+
+    // Use copyRanges to split the highlighted code into chunks,
+    // some of which are "live", i.e. copyable, some which are not.
+    // If there are no copyRanges, the whole snippet is copyable and there
+    // is no fancy live-chunk styling.
+    const mutableCopyRanges =
+      props.copyRanges !== undefined && props.copyRanges.slice();
+    let currentLiveRange = mutableCopyRanges && mutableCopyRanges.shift();
+    let currentChunk = [];
+    const allChunks = [];
+    const endCurrentChunk = ({ live }) => {
+      allChunks.push({
+        live,
+        highlightedLines: currentChunk.map(line => line.highlighted),
+        raw: currentChunk.reduce(
+          (result, line) => (result += line.raw + '\n'),
+          ''
+        ),
+        element: undefined
+      });
+      currentChunk = [];
+    };
+    for (let i = 0, l = splitDisplayCode.length; i < l; i++) {
+      const chunk = splitDisplayCode[i];
+      const lineNumber = i + 1;
+      if (currentLiveRange && lineNumber === currentLiveRange[0]) {
+        endCurrentChunk({ live: false });
+      } else if (currentLiveRange && lineNumber > currentLiveRange[1]) {
+        endCurrentChunk({ live: true });
+        currentLiveRange = mutableCopyRanges && mutableCopyRanges.shift();
+      }
+      currentChunk.push({
+        highlighted: chunk,
+        raw: rawCodeLines[i]
+      });
+    }
+    if (currentChunk.length) {
+      endCurrentChunk({ live: false });
+    }
+
+    const codeElements = [];
+    const highlightElements = [];
+    const copyElements = [];
+    let previousCount = 0;
+    let liveChunkCount = -1; // Incremented to give CopyButtons an identifier
+    allChunks.forEach((codeChunk, i) => {
+      const chunkId = `chunk-${i}`;
+
+      const lineEls = codeChunk.highlightedLines.map((line, i) => {
+        // Left padding is determined below
+        let lineClasses = 'pr12';
+        if (codeChunk.live) lineClasses += ' py3';
+        if (!codeChunk.live && props.copyRanges !== undefined)
+          lineClasses += ' opacity50 bg-darken10';
+
+        // Remove leading spaces, which are replaced with padding to avoid
+        // weird behaviors that occur when there are long unbroken strings:
+        // a line break might be introduced between the leading spaces and the
+        // long word, creating an empty line that nobody wanted.
+        const indentingSpacesMatch = line.match(/^[ ]*/);
+        const indentingSpaces = indentingSpacesMatch
+          ? indentingSpacesMatch[0]
+          : '';
+        const indentingSpacesCount = indentingSpaces.length;
+        const paddingLeft = indentingSpacesCount * props.characterWidth + 12;
+        const displayLine = line.replace(/^[ ]*/, '');
+
+        const lineNumberClasses = classnames(
+          'absolute fl color-gray-light align-r pr6',
```

Comment:
I think we could use color-gray and still achieve the same effect while adding a little more contrast:

```suggestion
          'absolute fl color-gray align-r pr6',
```